tests/ext/test_ratelimiter.py

```diff
@@ -358,3 +369,64 @@ async def test_group_caching(self, bot, intermediate):
         finally:
             TestAIORateLimiter.count = 0
             TestAIORateLimiter.call_times = []
+
+    async def test_allow_paid_broadcast(self, bot):
+        try:
+            rl_bot = ExtBot(
+                token=bot.token,
+                request=self.CountRequest(retry_after=None),
+                rate_limiter=AIORateLimiter(),
+            )
+
+            async with rl_bot:
+                apb_tasks = {}
+                non_apb_tasks = {}
+                for i in range(3000):
+                    apb_tasks[i] = asyncio.create_task(
+                        rl_bot.send_message(chat_id=-1, text="test", allow_paid_broadcast=True)
+                    )
+
+                number = 2
+                for i in range(number):
+                    non_apb_tasks[i] = asyncio.create_task(
+                        rl_bot.send_message(chat_id=-1, text="test")
+                    )
+                    non_apb_tasks[i + number] = asyncio.create_task(
+                        rl_bot.send_message(chat_id=-1, text="test", allow_paid_broadcast=False)
+                    )
+
+                await asyncio.sleep(0.85)
+                # We expect 5 non-apb requests:
+                # 1: `get_me` from `async with rl_bot`
+                # 2: `send_message` at time 0.00
+                # 3: `send_message` at time 0.25
+                # 4: `send_message` at time 0.50
+                # 5: `send_message` at time 0.75
```

Comment:
why are 5 messages spread out over such a long time period, is that normal?

Reply:
falsely copied from other test. you're right that the requests go through much faster due to initial burst. updated accordingly :)